event/event.go

```diff
@@ -239,11 +241,55 @@ type ReleaseEventCommon struct {
 	Error string `json:"error,omitempty"`
 }
 
+const (
+	// ReleaseImageSpecType is a type of release spec when there are update.Images
+	ReleaseImageSpecType = "releaseImageSpecType"
+	// ReleaseContainersSpecType is a type of release spec when there are update.Containers
+	ReleaseContainersSpecType = "releaseContainersSpecType"
+)
+
+// ReleaseSpec is a spec for images and containers release
+type ReleaseSpec struct {
+	// Type is ReleaseImageSpecType or ReleaseContainersSpecType
+	// if empty (for previous version), then use ReleaseImageSpecType
+	Type                  string
+	ReleaseImageSpec      *update.ReleaseImageSpec
+	ReleaseContainersSpec *update.ReleaseContainersSpec
+}
+
+// UnmarshalJSON for old version of spec (update.ReleaseImageSpec) where Type is empty
+func (s *ReleaseSpec) UnmarshalJSON(b []byte) error {
+	type T ReleaseSpec
+	t := (*T)(s)
+	if err := json.Unmarshal(b, t); err != nil {
+		return err
+	}
+
+	switch t.Type {
+	case "":
+		r := &update.ReleaseImageSpec{}
+		if err := json.Unmarshal(b, r); err != nil {
+			return err
+		}
+		s.Type = ReleaseImageSpecType
+		s.ReleaseImageSpec = r
+
+	case ReleaseImageSpecType, ReleaseContainersSpecType:
+		s.Type = t.Type
```

Comment:
This is tautological, since `t === s` :-) But it does make clear what's going on, and will probably be eliminated by the compiler.